editoast/src/views/infra/auto_fixes/mod.rs

```diff
@@ -182,6 +206,116 @@ fn fix_infra(
+            let UpdateOperation {
+                obj_id: id,
+                obj_type: r#type,
+                railjson_patch,
+                ..
+            } = update;
+            debug_assert_eq!(railjson.get_id(), &id);
+            debug_assert_eq!(railjson.get_type(), r#type);
+            serde_json::to_string(&*railjson)
+                .ok()
+                .and_then(|json| {
+                    let mut value = serde_json::from_str(&json).ok()?;
+                    json_patch::patch(&mut value, &railjson_patch).ok()?;
+                    Some(value)
+                })
+                .and_then(|value| serde_json::to_string(&value).ok())
+                .and_then(|json| serde_json::from_str(&json).ok())
+                .map(|railjson| Operation::Create(Box::new(railjson)))
+                .or_else(|| {
+                    error!("failed to apply patch when reducing CREATE+UPDATE");
+                    None
+                })
+        }
+        (Some(Operation::Delete(delete1)), Some(Operation::Delete(delete2))) => {
+            let DeleteOperation {
+                obj_id: id1,
+                obj_type: type1,
+            } = delete1;
+            let DeleteOperation {
+                obj_id: id2,
+                obj_type: type2,
+            } = delete2;
+            debug_assert_eq!(id1, id2);
+            debug_assert_eq!(type1, type2);
+            Some(Operation::Delete(DeleteOperation {
+                obj_id: id1,
+                obj_type: type1,
+            }))
+        }
+        (Some(Operation::Delete(delete1)), Some(Operation::Update(update2))) => {
+            let DeleteOperation {
+                obj_id: id1,
+                obj_type: type1,
+            } = delete1;
+            let UpdateOperation {
+                obj_id: id2,
+                obj_type: type2,
+                ..
+            } = update2;
+            debug_assert_eq!(id1, id2);
+            debug_assert_eq!(type1, type2);
+            Some(Operation::Delete(DeleteOperation {
+                obj_id: id1,
+                obj_type: type1,
+            }))
+        }
+        (Some(Operation::Update(update1)), Some(Operation::Update(update2))) => {
+            let UpdateOperation {
+                obj_id: id1,
+                obj_type: type1,
+                railjson_patch: mut patch1,
+            } = update1;
+            let UpdateOperation {
+                obj_id: id2,
+                obj_type: type2,
+                railjson_patch: patch2,
+            } = update2;
+            debug_assert_eq!(id1, id2);
+            debug_assert_eq!(type1, type2);
+            patch1.0.extend(patch2.0);
+            Some(Operation::Update(UpdateOperation {
+                obj_id: id1,
+                obj_type: type1,
+                railjson_patch: patch1,
+            }))
+        }
+        (Some(Operation::Update(update1)), Some(Operation::Delete(delete2))) => {
```

Comment:
It looks like a duplicate of (Delete, Update). We could avoid code duplication.

Reply:
Good point, done in https://github.com/osrd-project/osrd/pull/6012/commits/9b1fd0fb38d4b7f65acc0c867d2a382bc042a6ec.